Descent3/init.cpp

```diff
@@ -1988,8 +1986,19 @@ void SetupTempDirectory(void) {
   if (t_arg) {
     strcpy(Descent3_temp_directory, GameArgs[t_arg + 1]);
   } else {
-    // initialize it to custom/cache
-    ddio_MakePath(Descent3_temp_directory, Base_directory, "custom", "cache", NULL);
+    std::error_code ec;
+    std::filesystem::path tempPath = std::filesystem::temp_directory_path(ec);
+    if (ec) {
+      Error("Could not find temporary directory: \"%s\"", ec.message().c_str() );
+      exit(1);
+    }
+    ddio_MakePath(Descent3_temp_directory, tempPath.u8string().c_str(), "Descent3",
+                  "cache", NULL);
+    std::filesystem::create_directories(Descent3_temp_directory, ec);
+    if (ec) {
+      Error("Could not create temporary directory: \"%s\"", Descent3_temp_directory);
+      exit(1);
+    }
```

Comment:
Sorry, another request. This should be outside of `else` block, so we can try to create non-existent directory defined via `t_arg`.

Reply:
done too